### PR TITLE
feature: add context to state API

### PR DESCRIPTION
**Pull request by 1046102779**
Signed-off-by: 1046102779 <seachen@tencent.com>

## PR reference

https://github.com/dapr/components-contrib/pull/2208

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by 1046102779** [comment]
cc @daixiang0 @yaron2 @ItalyPaleAle

File: pkg/components/state/pluggable.go [hunk 1]
```diff
@@ -80,15 +81,15 @@ func (ss *grpcStateStore) Features() []state.Feature {
 }
 
 // Delete performs a delete operation.
-func (ss *grpcStateStore) Delete(req *state.DeleteRequest) error {
-	_, err := ss.Client.Delete(ss.Context, toDeleteRequest(req))
+func (ss *grpcStateStore) Delete(ctx context.Context, req *state.DeleteRequest) error {
+	_, err := ss.Client.Delete(ctx, toDeleteRequest(req))
```
Comment (ItalyPaleAle) [inline pkg/components/state/pluggable.go:85]: This doesn't use `ss.Context` anymore. Is that going to be a problem? CC: @mcandeia

File: pkg/components/state/pluggable_test.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func TestComponentCalls(t *testing.T) {
 		stStore, cleanup, err := getStateStore(svc)
 		require.NoError(t, err)
 		defer cleanup()
-		err = stStore.Delete(&state.DeleteRequest{
+		err = stStore.Delete(context.TODO(), &state.DeleteRequest{
```
Comment (ItalyPaleAle) [inline pkg/components/state/pluggable_test.go:240]: As is the case also in components-contrib, we use `context.Background()` in tests as this is not a TODO for the future.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
A few small changes please. Also there's one open question for Marcos to come in and advise on

File: pkg/components/state/pluggable_test.go (hunk 2 above)
Comment (1046102779) [reply pkg/components/state/pluggable_test.go]: Fixed.

File: go.mod [hunk 3]
```diff
@@ -402,3 +402,5 @@ replace (
 //
 // Then, run `make modtidy` in this repository.
 // This ensures that go.mod and go.sum are up-to-date.
+
+replace github.com/dapr/components-contrib => github.com/1046102779/components-contrib v1.6.0-rc.2.0.20221021113118-614a2d8c6543
```
[the remark this answers is not in the document]
Comment (1046102779) [reply go.mod:407]: Added.

File: pkg/components/state/pluggable.go (hunk 1 above)
Comment (1046102779) [reply pkg/components/state/pluggable.go:85]: image

I see that the member variable: `Context` in the GRPCConnector object is not assigned anywhere except after the initialization here, so I don't know what plans will be in the future.

Comment (ItalyPaleAle) [reply pkg/components/state/pluggable.go:85]: need to wait for @mcandeia  on that one

File: pkg/components/state/pluggable_test.go [hunk 4]
```diff
@@ -258,7 +258,7 @@ func TestComponentCalls(t *testing.T) {
...
```
Comment (ItalyPaleAle) [inline pkg/components/state/pluggable_test.go:261]: Please change all `context.TODO`'s in test (all test files) to `context.Background`

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: pkg/components/state/pluggable.go (hunk 1 above)
Comment (1046102779) [reply pkg/components/state/pluggable.go:85]: OK.

File: pkg/components/state/pluggable_test.go (hunk 4 above)
Comment (1046102779) [reply pkg/components/state/pluggable_test.go]: Sorry. I've just done an entire replacement using `sed` command.

**Review by ItalyPaleAle — DISMISSED** [review]

File: pkg/actors/actors.go [hunk 5]
```diff
@@ -1446,7 +1446,7 @@ func (a *actorsRuntime) getActorTypeMetadata(actorType string, migrate bool) (*A
 	} else {
 		retryErr = backoff.Retry(func() error {
 			metadataKey := constructCompositeKey("actors", actorType, "metadata")
-			resp, err := a.store.Get(&state.GetRequest{
+			resp, err := a.store.Get(context.TODO(), &state.GetRequest{
```
Comment (DeepanshuA) [inline pkg/actors/actors.go:1449]: Should we use context.Background() here as well?

Comment (ItalyPaleAle) [reply pkg/actors/actors.go:1449]: No, I think TODO here is correct. Because we will need to cascade the context, eventually.

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**